src/MongoDB.EntityFrameworkCore/Query/Visitors/MongoProjectionBindingRemovingExpressionVisitor.cs

```diff
@@ -65,12 +65,13 @@ protected override ProjectionExpression GetProjection(ProjectionBindingExpressio
     /// <inheritdoc />
     protected override Expression CreateGetValueExpression(
         Expression docExpression,
-        string? fieldName,
-        bool fieldRequired,
+        string? propertyName,
+        bool required,
         Type type,
+        ITypeBase? declaredType = null,
         CoreTypeMapping? typeMapping = null)
     {
-        var entityType = docExpression switch
+        var entityType = declaredType ?? docExpression switch
```

Comment:
What is declaredType?

Reply:
This is to allow the type discriminator to exist on the base type otherwise it looks for it on the concrete type.